broker-daemon/state-machines/fill-state-machine.js

```diff
@@ -214,9 +214,29 @@ const FillStateMachine = StateMachine.factory({
         throw new Error(`No engine available for ${outboundSymbol}`)
       }
 
-      const [ feeRefundPaymentRequest, depositRefundPaymentRequest ] = await Promise.all([
-        payInvoice(feeRequired, feePaymentRequest, outboundEngine, this.logger, 'fee', fillId),
-        payInvoice(depositRequired, depositPaymentRequest, outboundEngine, this.logger, 'deposit', fillId)
+      this.logger.debug(`Paying fee and deposit invoices for ${fillId}`)
+
+      let payFeeInvoice
+      let payDepositInvoice
+
+      if (feeRequired) {
+        payFeeInvoice = payInvoice(outboundEngine, feePaymentRequest)
+      } else {
+        this.logger.debug(`Skipping paying fee invoice for ${fillId}, not required`)
+      }
+
+      if (depositRequired) {
+        payDepositInvoice = payInvoice(outboundEngine, depositPaymentRequest)
+      } else {
+        this.logger.debug(`Skipping paying deposit invoice for ${fillId}, not required`)
+      }
+
+      const [
+        feeRefundPaymentRequest,
+        depositRefundPaymentRequest
```

Comment:
This is a question for below, but will the call to `fillOrder` fail if these are undefined (meaning they are not provided in the request to the relayer)?

Reply:
fillOrder won't fail if these are undefined (so long as the fee/deposit was not required)